recovery/device.go

```diff
@@ -115,18 +114,16 @@ func (d *DiskDevice) CreateLUKSPartition(size uint64, label string, keyfile stri
 	// FIXME: determine partition name in a civilized way
 	partdev := d.partDev(4)
 
-	// Set up LUKS device
-	logger.Noticef("Create LUKS keyfile")
-	buffer := make([]byte, keyfileSize)
-	rand.Read(buffer)
-	if err := ioutil.WriteFile(keyfile, buffer, 0400); err != nil {
-		return fmt.Errorf("cannot create keyfile %s: %s", keyfile, err)
-	}
-
-	// Don't remove this delay, prevents kernel crash
+	// Don't remove this delay, it prevents a kernel crash
 	// see https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1835279
 	time.Sleep(1 * time.Second)
 
+	// Create a temporary unsealed keyfile on a RAM-backed filesystem
```

Comment:
You could instead use `-` as keyfile and write the key to standard input of the process. I think this is marginally more secure as there is nothing in the filesystem that ever records it.

Reply:
Agreed, will fix that.